src/jdk.jfr/share/classes/jdk/jfr/internal/RequestEngine.java

```diff
@@ -191,20 +191,21 @@ static void doChunkBegin() {
     }
 
     private static void doChunk(Predicate<PlatformEventType> predicate) {
+        long timestamp = JVM.counterTime();
         for (RequestHook requestHook : entries) {
             PlatformEventType s = requestHook.type;
             if (s.isEnabled() && predicate.test(s)) {
-                requestHook.execute();
+                requestHook.execute(timestamp);
             }
         }
     }
 
     static long doPeriodic() {
-        return run_requests(entries);
+        return run_requests(entries, JVM.counterTime());
     }
 
     // code copied from native impl.
-    private static long run_requests(Collection<RequestHook> entries) {
+    private static long run_requests(Collection<RequestHook> entries, long timestamp) {
         long last = lastTimeMillis;
         // Bug 9000556 - current time millis has rather lame resolution
         // The use of os::elapsed_counter() is deliberate here, we don't
```

Comment:
While you are here can you delete this comment block, or suitably modify it so that it makes sense for the code that follows. Thanks.

Reply:
Actually what is all this `lastTimeMillis` code for? And how does it relate to the passed in timestamp?

Reply:
In Oracle JDK 8, the loop for emitting periodic event was written in native. With the rewrite that happened in JDK 9, the native logic was moved to Java, but to ensure the the mechanism worked the same, the implementation was largely kept as is. Therefore the name run_requests(...). 

The timestamps in that method, for example lastTimeMillis, make sure events are emitted at the correct time, for example, one event could have the period 5s and the other 7s.

The timestamp I added is a timestamp that could be used when committing the event in native, for example:

     EventJVMInformation event(UNTIMED);
     event.set_jvmName(VM_Version::vm_name());
     event.set_endtime(timestamp());
     event.commit();

I can see if I can make the comment and naming of the variable more clear.